analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -240,7 +240,14 @@ context.Operation.Instance.Kind switch
         private static ExceptionState ThrowExceptionType(IOperation operation) =>
             operation.Kind switch
             {
+                OperationKindEx.ArrayElementReference => new ExceptionState(operation.Type),
+                OperationKindEx.FieldReference => new ExceptionState(operation.Type),
+                OperationKindEx.Invocation => new ExceptionState(operation.Type),
+                OperationKindEx.LocalReference => new ExceptionState(operation.Type),
                 OperationKindEx.ObjectCreation => new ExceptionState(operation.Type),
+                OperationKindEx.ParameterReference => new ExceptionState(operation.Type),
+                OperationKindEx.PropertyReference => new ExceptionState(operation.Type),
+                OperationKindEx.Conversion => new ExceptionState(IConversionOperationWrapper.FromOperation(operation).Operand.Type),
```

Comment:
What about recursion? To support (not so smart) stuff like `throw (System.IO.IOException)new System.IO.FileNotFoundException();` that contains explicit conversion and implicit conversion.
```suggestion
                OperationKindEx.Conversion => ThrowExceptionType(IConversionOperationWrapper.FromOperation(operation).Operand.Type),
```

Reply:
I guess this does the trick. All the tests pass. I'll add also one more with your example.